### PR TITLE
issue #202 ability to provide options to the MomentModule

### PR DESCRIPTION
**Pull request by StickNitro**
Add ability to provide options to the moment module using the `forRoot` convention to supply options globally or per pipe

### CODE REVIEW

File: src/duration.pipe.ts [hunk 1]
```diff
@@ -1,12 +1,36 @@
...
+
   transform(value: any, ...args: string[]): string {
     if (typeof args === 'undefined' || args.length !== 1) {
       throw new Error('DurationPipe: missing required time unit argument');
     }
     return moment.duration(value, args[0] as moment.unitOfTime.DurationConstructor).humanize();
   }
+
+  private _applyOptions(momentOptions: NgxMomentOptions): void {
+    if (!!!momentOptions) {
```
Comment (urish) [inline src/duration.pipe.ts]: Why triple negation? I'd rather use single negation

File: src/duration.pipe.spec.ts [hunk 2]
```diff
@@ -17,5 +18,23 @@ describe('DurationPipe', () => {
...
+    const momentOptions: NgxMomentOptions = {
+      relativeTimeThresholdOptions: {
+        'm': 59
+      }
+    };
+
+    beforeEach(() => pipe = new DurationPipe(momentOptions));
+
+    it(`should convert '50 minutes' to '50 minutes' when relativeTimeThreshold for 'm' unit is set to 59`, () => {
+      expect(pipe.transform(50, 'minutes')).toEqual('50 minutes');
```
Comment (urish) [inline src/duration.pipe.spec.ts]: 👍

File: src/moment-options.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+export interface NgxMomentOptions {
+  /**
+   * relativeTimeThresholdOptions
+   * @description Provides the `relativeTimeThreshold` units allowing a pipe to set the `moment.relativeTimeThreshold` values.
+   * The `key` is a unit defined as one of `ss`, `s`, `m`, `h`, `d`, `M`.
+   * @see https://momentjs.com/docs/#/customization/relative-time-threshold/
+   * @example by default more than 45 seconds is considered a minute, more than 22 hours is considered a day and so on.
+   * So settings the unit 'm' to `59` will adjust the `relativeTimeThreshold` and consider more than 59 minutes
+   * to be an hour (default is `45 minutes`)
+   */
```
Comment (urish) [inline src/moment-options.ts]: Thanks for adding this doc comment. Can you please also add a note to the README explaining this?

File: src/moment-options.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { InjectionToken } from '@angular/core';
+
+export const NGX_MOMENT_OPTIONS: InjectionToken<string> = new InjectionToken<string>('NGX_MOMENT_OPTIONS');
```
Comment (urish) [inline src/moment-options.ts]: According to docs, it seems like the type parameter for `InjectionToken` should be the interface name (i.e. `NgxMomentOptions`):

https://angular.io/api/core/InjectionToken

File: src/duration.pipe.ts (hunk 1 above)
Comment (StickNitro) [reply src/duration.pipe.ts]: I felt triple negation was safer, equivalent to `if (!(!!momentOptions))`.  If you prefer single negation ill make the change

File: src/moment-options.ts (hunk 4 above)
Comment (StickNitro) [reply src/moment-options.ts]: Yep, my bad and good spot, updated in next commit.

File: src/moment-options.ts (hunk 3 above)
Comment (StickNitro) [reply src/moment-options.ts]: README updated in next commit

File: src/duration.pipe.ts (hunk 1 above)
Comment (StickNitro) [reply src/duration.pipe.ts]: change made

**Comment by urish** [comment]
Thank you!

**Comment by deeg** [comment]
@urish , also looking for this. Any chance you could cut a new release to include this?

**Comment by urish** [comment]
#deeg I plan to, when I get a chance, I'm currently on a long vacation in Japan

**Comment by urish** [comment]
@deeg I plan to, when I get a chance, I'm currently on a long vacation in Japan

**Comment by deeg** [comment]
@urish , no rush, have fun on the vacation! I hear soon all the cherry blossoms will be in full force. Will you get to enjoy that?

**Comment by urish** [comment]
@deeg Hopefully so! You can read about our journey so far in [my blog post](https://medium.com/@urish/first-week-in-japan-3f7de32df56d).

Regarding this PR - released as part of 3.4.0

**Pull request merged**